newIDE/app/src/MainFrame/EditorTabs/DraggableEditorTabs.js

```diff
@@ -0,0 +1,155 @@
+// @flow
+import * as React from 'react';
+
+import { makeDragSourceAndDropTarget } from '../../UI/DragAndDrop/DragSourceAndDropTarget';
+import { ScreenTypeMeasurer } from '../../UI/Reponsive/ScreenTypeMeasurer';
+import { ColumnDropIndicator } from './DropIndicator';
+import {
+  type EditorTabsState,
+  type EditorTab,
+  getEditors,
+  getCurrentTabIndex,
+} from './EditorTabsHandler';
+import {
+  ClosableTabs,
+  ClosableTab,
+  type ClosableTabProps,
+} from '../../UI/ClosableTabs';
+
+type DraggableClosableTabsProps = {|
+  hideLabels?: boolean,
+  editorTabs: EditorTabsState,
+  onClickTab: (index: number) => void,
+  onCloseTab: (editor: EditorTab) => void,
+  onCloseOtherTabs: (editor: EditorTab) => void,
+  onCloseAll: () => void,
+  onTabActive: (editor: EditorTab) => void,
+  onDropTab: (fromIndex: number, toIndex: number) => void,
+|};
+
+export function DraggableClosableTabs({
+  hideLabels,
+  editorTabs,
+  onClickTab,
+  onCloseTab,
+  onCloseOtherTabs,
+  onCloseAll,
+  onTabActive,
+  onDropTab,
+}: DraggableClosableTabsProps) {
+  let draggedTabIndex: number | typeof undefined = undefined;
+
+  return (
+    <ClosableTabs hideLabels={hideLabels}>
+      {getEditors(editorTabs).map((editorTab, id) => {
+        const isCurrentTab = getCurrentTabIndex(editorTabs) === id;
+        return (
+          <DraggableClosableTab
+            index={id}
+            label={editorTab.label}
+            key={editorTab.key}
+            id={`tab-${editorTab.key.replace(/\s/g, '-')}`}
+            active={isCurrentTab}
+            onClick={() => onClickTab(id)}
+            onClose={() => onCloseTab(editorTab)}
+            onCloseOthers={() => onCloseOtherTabs(editorTab)}
+            onCloseAll={onCloseAll}
+            onActivated={() => onTabActive(editorTab)}
+            closable={editorTab.closable}
+            onBeginDrag={() => {
+              draggedTabIndex = id;
+              return editorTab;
+            }}
+            onDrop={toIndex => {
+              if (draggedTabIndex !== undefined) {
```

Comment:
I think that you don't need your index to be undefined, you can initialise it with `null` and set it to `null` just below.
This check could become `if (typeof draggedTabIndex === number)` (to avoid being confused with 0)